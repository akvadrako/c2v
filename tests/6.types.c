#include <stdio.h>
#include <stddef.h>
#include <stdbool.h>
#include <stdint.h>

// standard types should be lower-cased
typedef unsigned short ushort;
typedef float f32;
typedef size_t size_t;
typedef void *null;
typedef size_t usize;
typedef ptrdiff_t isize;
typedef bool maybe;

// not sure how this is supposed to work
typedef intptr_t iptr;

int main()
{
    void *pointers[8];

    return 0;
}
