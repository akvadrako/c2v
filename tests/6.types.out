[translated]
module main

type Wchar_t = int
type Ushort = u16
type F32 = f32
type Null = voidptr
type Usize = usize
type Isize = isize
type Maybe = bool
type Iptr = C.intptr_t

fn main() {
	pointers := [8]voidptr{}
	return
}
